backend/functions/src/api/resources/auth/auth.model.ts

```diff
@@ -0,0 +1,32 @@
+import { IsAlpha, IsAlphanumeric, IsDefined, IsEmail, IsNotEmpty, Length } from 'class-validator';
+
+export class SignupRequestPayLoad {
+  @Length(8, 20)
+  @IsNotEmpty()
+  @IsDefined({ message: 'Username is missing' })
```

Comment:
`@IsDefined()` checks if the value is not `null` or `undefined`. `@IsEmpty()` checks that the value is not `null` or `undefined` or `""`. Since `@IsEmpty()` does a more exhaustive check, using `@IsDefined()` is unnecessary.

For all fields in both classes, **IF** you are using **BOTH** of these, remove `@IsDefined()`.

Reply:
That was just for different messages